File: AppliSportSpringBoot/src/main/java/projet/AppliSport/model/Caracteristique.java
package projet.AppliSport.model;

import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.validation.constraints.Max;
import javax.validation.constraints.Min;

import com.fasterxml.jackson.annotation.JsonView;

import projet.AppliSport.views.Views;




@Entity
@Table(name = "caracteristique")
@SequenceGenerator(name = "seqCaracteristique", sequenceName = "seq_caracteristique", initialValue = 1, allocationSize = 1)
public class Caracteristique {

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqCaracteristique")
	@Column(name = "carac_id")
	@JsonView(Views.Common.class)
	private Long id;
	
	@Min(value=0)
	@Max(value=10)
	@Column(name = "carac_mentale_collectif")
	@JsonView(Views.Common.class)
	private int collectif;
	
	@Min(value=0)
	@Max(value=10)
	@Column(name = "carac_mentale_creativite")
	@JsonView(Views.Common.class)
	private int creativite;
	
	@Min(value=0)
	@Max(value=10)
	@Column(name = "carac_mentale_determination")
	@JsonView(Views.Common.class)
	private int determination;
	
	@Min(value=0)
	@Max(value=10)
	@Column(name = "carac_mentale_patience")
	@JsonView(Views.Common.class)
	private int patience;
	
	@Min(value=0)
	@Max(value=10)
	@Column(name = "carac_physique_agilite")
	@JsonView(Views.Common.class)
	private int detente;
	
	@Min(value=0)
	@Max(value=10)
	@Column(name = "carac_physique_detente")
	@JsonView(Views.Common.class)
	private int agilite;
	
	@Min(value=0)
	@Max(value=10)
	@Column(name = "carac_physique_puissance")
	@JsonView(Views.Common.class)
	private int puissance;
	
	@Min(value=0)
	@Max(value=10)
	@Column(name = "carac_physique_vitesse")
	@JsonView(Views.Common.class)
	private int vitesse;
	
	@Min(value=0)
	@Max(value=10)
	@Column(name = "carac_physique_endurance")
	@JsonView(Views.Common.class)
	private int endurance;

	
	@OneToOne(mappedBy ="caracteristique")
	@JsonView(Views.CaracteristiqueWithProfil.class)
	private Profil profil;


	public Caracteristique() {
	}

	public Caracteristique(int collectif, int creativite, int determination, int patience, int detente, int agilite,
			int puissance, int vitesse, int endurance) {
		super();
		this.collectif = collectif;
		this.creativite = creativite;
		this.determination = determination;
		this.patience = patience;
		this.detente = detente;
		this.agilite = agilite;
		this.puissance = puissance;
		this.vitesse = vitesse;
		this.endurance = endurance;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public int getCollectif() {
		return collectif;
	}

	public int getCreativite() {
		return creativite;
	}

	public int getDetermination() {
		return determination;
	}

	public int getPatience() {
		return patience;
	}

	public int getDetente() {
		return detente;
	}

	public int getPuissance() {
		return puissance;
	}

	public int getVitesse() {
		return vitesse;
	}

	public int getEndurance() {
		return endurance;
	}

	public void setCollectif(int collectif) {
		this.collectif = collectif;
	}

	public void setCreativite(int creativite) {
		this.creativite = creativite;
	}

	public void setDetermination(int determination) {
		this.determination = determination;
	}

	public void setPatience(int patience) {
		this.patience = patience;
	}

	public void setDetente(int detente) {
		this.detente = detente;
	}

	public void setPuissance(int puissance) {
		this.puissance = puissance;
	}

	public void setVitesse(int vitesse) {
		this.vitesse = vitesse;
	}

	public void setEndurance(int endurance) {
		this.endurance = endurance;
	}

	public int getAgilite() {
		return agilite;
	}

	public void setAgilite(int agilite) {
		this.agilite = agilite;
	}

	public Profil getProfil() {
		return profil;
	}

	public void setProfil(Profil profil) {
		this.profil = profil;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Caracteristique other = (Caracteristique) obj;
		return Objects.equals(id, other.id);
	}

}

File: SportifyApp/src/app/model/caracteristique.ts
import { Sport } from './sport';
import { Profil } from './profil';

export class Caracteristique {
  private _id: number | undefined;
  private _collectif: number | undefined;
  private _creativite: number | undefined;
  private _determination: number | undefined;
  private _patience: number | undefined;
  private _detente: number | undefined;
  private _agilite: number | undefined;
  private _puissance: number | undefined;
  private _vitesse: number | undefined;
  private _endurance: number | undefined;
  private _profil: Profil | undefined;
  private _sport: Sport | undefined;

  constructor(
    id?: number,
    collectif?: number,
    creativite?: number,
    determination?: number,
    patience?: number,
    detente?: number,
    agilite?: number,
    puissance?: number,
    vitesse?: number,
    endurance?: number,
    profil?: Profil,
    sport?: Sport
  ) {
    this._id = id;
    this._collectif = collectif;
    this._creativite = creativite;
    this._determination = determination;
    this._patience = patience;
    this._detente = detente;
    this._agilite = agilite;
    this._puissance = puissance;
    this._vitesse = vitesse;
    this._endurance = endurance;
    this._profil = profil;
    this._sport = sport;
  }

  /**
   * Getter id
   * @return {number }
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter collectif
   * @return {number }
   */
  public get collectif(): number | undefined {
    return this._collectif;
  }

  /**
   * Getter creativite
   * @return {number }
   */
  public get creativite(): number | undefined {
    return this._creativite;
  }

  /**
   * Getter determination
   * @return {number }
   */
  public get determination(): number | undefined {
    return this._determination;
  }

  /**
   * Getter patience
   * @return {number }
   */
  public get patience(): number | undefined {
    return this._patience;
  }

  /**
   * Getter detente
   * @return {number }
   */
  public get detente(): number | undefined {
    return this._detente;
  }

  /**
   * Getter agilite
   * @return {number }
   */
  public get agilite(): number | undefined {
    return this._agilite;
  }

  /**
   * Getter puissance
   * @return {number }
   */
  public get puissance(): number | undefined {
    return this._puissance;
  }

  /**
   * Getter vitesse
   * @return {number }
   */
  public get vitesse(): number | undefined {
    return this._vitesse;
  }

  /**
   * Getter endurance
   * @return {number }
   */
  public get endurance(): number | undefined {
    return this._endurance;
  }

  /**
   * Getter profil
   * @return {Profil }
   */
  public get profil(): Profil | undefined {
    return this._profil;
  }

  /**
   * Setter id
   * @param {number } value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter collectif
   * @param {number } value
   */
  public set collectif(value: number | undefined) {
    this._collectif = value;
  }

  /**
   * Setter creativite
   * @param {number } value
   */
  public set creativite(value: number | undefined) {
    this._creativite = value;
  }

  /**
   * Setter determination
   * @param {number } value
   */
  public set determination(value: number | undefined) {
    this._determination = value;
  }

  /**
   * Setter patience
   * @param {number } value
   */
  public set patience(value: number | undefined) {
    this._patience = value;
  }

  /**
   * Setter detente
   * @param {number } value
   */
  public set detente(value: number | undefined) {
    this._detente = value;
  }

  /**
   * Setter agilite
   * @param {number } value
   */
  public set agilite(value: number | undefined) {
    this._agilite = value;
  }

  /**
   * Setter puissance
   * @param {number } value
   */
  public set puissance(value: number | undefined) {
    this._puissance = value;
  }

  /**
   * Setter vitesse
   * @param {number } value
   */
  public set vitesse(value: number | undefined) {
    this._vitesse = value;
  }

  /**
   * Setter endurance
   * @param {number } value
   */
  public set endurance(value: number | undefined) {
    this._endurance = value;
  }

  /**
   * Setter profil
   * @param {Profil } value
   */
  public set profil(value: Profil | undefined) {
    this._profil = value;
  }

  /**
   * Getter sport
   * @return {Sport }
   */
  public get sport(): Sport | undefined {
    return this._sport;
  }

  /**
   * Setter sport
   * @param {Sport } value
   */
  public set sport(value: Sport | undefined) {
    this._sport = value;
  }
}
